agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,54 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpu_stats
+    # @param [Array<Vmstat::Cpu>] current_cpu_stats
+    # @return [Hash]
+    def calculate_average_cpu(prev_cpu_stats, current_cpu_stats)
+      all = prev_cpu_stats.zip(current_cpu_stats).map do |prev, current|
```

Comment:
Yes thats correct.  I actually found someone online doing something similar w/ VmStat data, that's how I came to this.  Pretty cool once you wrap your head around it.